src/main/java/com/icthh/xm/ms/balance/repository/BalanceRepository.java

```diff
@@ -14,4 +24,14 @@
 @Repository
 public interface BalanceRepository extends JpaRepository<Balance, Long>, JpaSpecificationExecutor<Balance>, ResourceRepository {
 
+    @Query("SELECT SUM(p.amount) FROM Pocket as p WHERE p.balance = :balance AND ((p.startDateTime < :today) OR (p.startDateTime IS NULL)) AND ((p.endDateTime > :today) OR (p.endDateTime IS NULL))")
+    Optional<BigDecimal> getBalanceAmount(@Param("balance") Balance balance, @Param("today") Instant today);
```

Comment:
what is the purpose of the parameter 'today'? 
if it indicates only `current date` then may be we can omit it at all
if it provides possibility to shift start date then it may be better to name it `startDateTime`

Reply:
today it's now
we can omit it at all in jpql, a know how do it in sql but NOW() now working in jpql?